src/modules/new-request-form/useEndUserConditions.tsx

```diff
@@ -0,0 +1,21 @@
+import type { EndUserCondition, Field } from "./data-types";
+
+export function useEndUserConditions(
+  fields: Field[],
+  endUserConditions: EndUserCondition[]
+): Field[] {
+  return fields.filter((field) => {
+    const conditions = endUserConditions.filter((condition) =>
+      condition.child_fields.some((childField) => childField.id === field.id)
+    );
+
+    const hasNoConditions = conditions.length === 0;
+    const meetsAnyCondition = conditions.some(
+      (condition) =>
+        fields.find((field) => field.id === condition.parent_field_id)
+          ?.value === condition.value
+    );
+
+    return hasNoConditions || meetsAnyCondition;
```

Comment:
Do we need to check for `hasNoConditions`? Shouldn't `meetsAnyCondition` be false if `conditions` is empty?

Reply:
> Shouldn't `meetsAnyCondition` be false if `conditions` is empty?

Yes and that is the problem because we want the return value to still be `true`.
So if `conditions` is empty, `hasNoConditions` is `true` and `meetsAnyCondition` is `false`. The return value will be `true` || `false` which is `true`.
If we just use `meetsAnyCondition` than it is always `false` so it will only show `fields` with `conditions`.